rules/sairedis.dep

```diff
@@ -0,0 +1,17 @@
+
+#DPKG FRK
+SPATH       := $($(LIBSAIREDIS)_SRC_PATH)
+DEP_FILES   := $(SONIC_COMMON_FILES_LIST) rules/sairedis.mk rules/sairedis.dep   
+DEP_FILES   += $(SONIC_COMMON_BASE_FILES_LIST)
+
+SMDEP_PATHS := $(SPATH) $(SPATH)/SAI $(SPATH)/SAI/bm/behavioral-model $(SPATH)/SAI/test/ptf $(SPATH)/SAI/test/saithrift/ctypesgen
+$(foreach path, $(SMDEP_PATHS), $(eval $(path) :=$(filter-out $(SMDEP_PATHS),$(addprefix $(path)/, \
+					$(shell cd $(path)  && git ls-files | grep -v " ")))))
```

Comment:
why do we have such complexity here? can you help to explain?

Reply:
I do not really understand SMDEP has these ```$(SPATH)/SAI/bm/behavioral-model $(SPATH)/SAI/test/ptf $(SPATH)/SAI/test/saithrift/ctypesgen```

and I see this error
```
fatal: Not a git repository: /data2/sonic/stretch/sonic-buildimage/.git/modules/sonic-sairedis/modules/SAI/modules/bm/behavioral-model
```

Reply:
Fixed and uploaded the patchset.

Reply:
This package has multiple Sub repo. The git hash-object command is not working for the repo directory name. Hence we need to exclude the directory name.  We also need to exclude the files that has space on it.